### ISSUE TITLE
Using #set does not persist changed values

### ISSUE THREAD

**Issue by aarongodin** [anykeyh/clear#81]
Hi again :) After reviewing the inline docs, I found the `#set` method on `HasColumns`. I made an assumption that this method was what I needed to update many columns at once with a hash. The behavior I'm seeing is that the values are not queued for saving, and calling `#save!` does not query postgres to perform an update.

Example code:

```crystal
require "clear"

require "./src/db"

Lizzy::DB.connect

module Lizzy::Groups
  class GroupModel
    include Clear::Model

    self.table = "groups"

    column group_id : String, primary: true, presence: false
    column tenant_id : String
    column name : String
  end
end

group = Lizzy::Groups::GroupModel.find!("b465207a-09e7-4a29-9c65-9a69e359fc86")

pp "Before calling set"
pp group.update_h

group.set({
  name: "testing the set method"
})

pp "After calling set"
pp group.update_h

group.save!

pp "After calling save"
pp group
```

The output from running this example:

```
14:15 $ crystal run model_test.cr
"Before calling set"
{}
"After calling set"
{}
"After calling save"
#<Lizzy::Groups::GroupModel:0x7f15ed1ecb90
 @attributes={},
 @cache=
  #<Clear::Model::QueryCache:0x7f15ed1e2480
   @cache={},
   @cache_activation=Set{}>,
 @errors=[],
 @group_id_column="b465207a-09e7-4a29-9c65-9a69e359fc86",
 @name_column="testing the set method",
 @persisted=true,
 @tenant_id_column="c44ddd8d-5b36-445d-af79-e05107c992c8">
```

Is my intention correct on how `#set` is intended to be used? Setting the `name` column via `group.name = "new name"` is working.

**Comment by anykeyh** [comment]
Hello !

`#set` is used internally when we call initialize on the models. So it behave without turning the `dirty` flags on for a good reason.
However I see your current use and I think Clear is wrong, because there's no other methods to populate a model from an hash or tuple and trigger the dirty flag of the column.

I'll fix this asap, by defining two sets of methods, `set` (values changed become dirty) and `reset` (value changed remains clean).

**Comment by anykeyh** [comment]
This has been fixed in 0.6. current `set` behavior has been renamed `reset`, while `set` now will set the `changed?` flag to true. You have nothing to do (except update of Clear)

**Comment by aarongodin** [comment]
Awesome, thanks for the quick turnaround!
